### PR TITLE
Chore: update README with table of contents

### PR DESCRIPTION
**Pull request by thisisobate**

### CODE REVIEW

File: programs/lfx-mentorship/2024/01-Mar-May/README.md [hunk 1]
```diff
@@ -28,8 +28,95 @@ Mentee application instructions can be found on the [Program Guidelines](https:/
 ---
 
 ## Table of Contents
+- [Term 01 - 2024 March - May](#term-01---2024-march---may)
+    - [Timeline](#timeline)
+    - [Project Instructions](#project-instructions)
+    - [Application instructions](#application-instructions)
+  - [Table of Contents](#table-of-contents)
+  - [Accepted Projects](#accepted-projects)
```
Comment (nate-double-u) [inline programs/lfx-mentorship/2024/01-Mar-May/README.md:36]: Could we drop this section? All we need for the ToC here is the list of projects (everything else is above the ToC so should be fairly easy to find 🙂)

File: programs/lfx-mentorship/2024/01-Mar-May/README.md [hunk 2]
```diff
@@ -28,8 +28,95 @@ Mentee application instructions can be found on the [Program Guidelines](https:/
...
+    - [Volcano](#volcano)
+      - [Volcano supports multi-cluster AI workloads scheduling](#volcano-supports-multi-cluster-ai-workloads-scheduling)
+      - [Volcano supports DRA integration](#volcano-supports-dra-integration)
+    - [WasmEdge](#wasmedge)
+      - [Integrate MLX as a new WASI-NN backend](#integrate-mlx-as-a-new-wasi-nn-backend)
+      - [Integrate Intel Extension for Transformers as a new WASI-NN backend](#integrate-intel-extension-for-transformers-as-a-new-wasi-nn-backend)
+      - [Integrate whisper.cpp as a new WASI-NN backend](#integrate-whispercpp-as-a-new-wasi-nn-backend)
+      - [Integrate burn.rs as a new WASI-NN backend](#integrate-burnrs-as-a-new-wasi-nn-backend)
 
----
```
Comment (nate-double-u) [inline programs/lfx-mentorship/2024/01-Mar-May/README.md:32]: let's not drop this line.

File: programs/lfx-mentorship/2024/01-Mar-May/README.md (hunk 1 above)
Comment (thisisobate) [reply programs/lfx-mentorship/2024/01-Mar-May/README.md:36]: Alright 👍

File: programs/lfx-mentorship/2024/01-Mar-May/README.md (hunk 2 above)
Comment (thisisobate) [reply programs/lfx-mentorship/2024/01-Mar-May/README.md:32]: Alright 👍

**Review by nate-double-u — APPROVED** [review]
Thanks Uche!

**Pull request merged**